yugabyte/yugabyte-db
# [#7706] Platform: Edit backup config credentials.

**Description:**
Backup config security credentials can change over time. The platform should support editing the security credentials even if there are existing backups associated with a Backup Config. When security credentials expire, the Platform will fail to take new backups and also fail to delete old backups.

This change should be implemented for all three storage configs:

AWS: Allow editing Access Key/Secret, also toggling between IAM and AccessKey
GCP: Allow editing GCS Credentials [Do not allow editing bucket name]
Azure: Allow editing Azure SAS token [Do not allow editing Container URL]

**Test Plan:**

1. Hit the config button from the SideNav.
2. Click on Backup Tab and then hit the `amazon s3` tab.
3. Create a backup config for s3.
4. Once it got created then you'll see the Edit configuration button next to the Delete configuration button.
5. Click on the Edit Configuration button.
6. AWS access credentials should be editable. 

Do the same for the rest of the configs.

**SBT test**

`[info] Passed: Total 1304, Failed 0, Errors 0, Passed 1300, Skipped 4
[success] Total time: 1187 s (19:47), completed 12 May, 2021 3:55:27 PM`
